src/app/wallets/get-lightning-fee.ts

```diff
@@ -0,0 +1,161 @@
+
+  const { amount: lnInvoiceAmount } = decodedInvoice
+  if (lnInvoiceAmount && lnInvoiceAmount > 0) {
+    return new LnPaymentRequestZeroAmountRequiredError()
+  }
+
+  const paymentAmount = checkedToSats(amount)
+  if (paymentAmount instanceof Error) return paymentAmount
+
+  return noAmountProbeForFee({ walletPublicId, decodedInvoice, paymentAmount, logger })
+}
+
+const feeProbe = async ({
+  walletPublicId,
+  decodedInvoice,
+  paymentAmount,
+  logger,
+}: {
+  walletPublicId: WalletPublicId
+  decodedInvoice: LnInvoice
+  paymentAmount: Satoshis
+  logger: Logger
+}): Promise<Satoshis | ApplicationError> => {
+  const { destination, paymentHash, milliSatsAmount } = decodedInvoice
+
+  const walletId = await walletIdFromPublicId(walletPublicId)
+  if (walletId instanceof Error) return walletId
+
+  const balance = await getBalanceForWallet({ walletId, logger })
+  if (balance instanceof Error) return balance
+  if (balance < paymentAmount) {
+    return new InsufficientBalanceError(
+      `Payment amount '${paymentAmount}' exceeds balance '${balance}'`,
+    )
+  }
+
+  const lndService = LndService()
+  if (lndService instanceof Error) throw lndService
+  if (lndService.isLocal(destination)) {
+    return toSats(0)
+  }
+
+  const key = CachedRouteLookupKeyFactory().create({
+    paymentHash,
+    milliSats: milliSatsAmount,
+  })
+  const routeFromCache = await RoutesCache().findByKey(key)
+  const validCachedRoute = !(routeFromCache instanceof Error)
+  if (validCachedRoute) return toSats(routeFromCache.route.fee)
+
+  const maxFee = LnFeeCalculator().max(paymentAmount)
+
+  const rawRoute = await lndService.findRouteForInvoice({ decodedInvoice, maxFee })
+  if (rawRoute instanceof Error) return rawRoute
+
+  const routeToCache = { pubkey: lndService.defaultPubkey(), route: rawRoute }
+  const cachedRoute = await RoutesCache().store({ key, routeToCache })
+  if (cachedRoute instanceof Error) return cachedRoute
+
+  return toSats(rawRoute.fee)
+}
+
+const noAmountProbeForFee = async ({
+  walletPublicId,
+  decodedInvoice,
+  paymentAmount,
+  logger,
+}: {
+  walletPublicId: WalletPublicId
+  decodedInvoice: LnInvoice
+  paymentAmount: Satoshis
+  logger: Logger
+}): Promise<Satoshis | ApplicationError> => {
+  const { destination, paymentHash } = decodedInvoice
+
+  const walletId = await walletIdFromPublicId(walletPublicId)
+  if (walletId instanceof Error) return walletId
+
+  const balance = await getBalanceForWallet({ walletId, logger })
+  if (balance instanceof Error) return balance
+  if (balance < paymentAmount) {
+    return new InsufficientBalanceError(
+      `Payment amount '${paymentAmount}' exceeds balance '${balance}'`,
+    )
+  }
+
+  const lndService = LndService()
+  if (lndService instanceof Error) throw lndService
+  if (lndService.isLocal(destination)) {
+    return toSats(0)
+  }
+
+  const key = CachedRouteLookupKeyFactory().create({
+    paymentHash,
+    milliSats: toMilliSatsFromNumber(paymentAmount),
```

Comment:
what is the associated path for noInvoicePayment that is using the cached route?

a quick search only lead me to https://github.com/GaloyMoney/galoy/blob/b98adfcb4cdbb607cac0bc913ba6274bbe0c0660/src/app/wallets/ln-send-payment.ts#L464, but this assume the invoice is not amountLess.

Could this be the issue I have seen regarding the cache/probe that is not been used when I try paying an invoice? (and maybe you only try reproducing with invoiceFull amount?)

Reply:
Oh this is a good catch, we should ideally be using `amount` there instead of `decodedInvoice.milliSatsAmount` to also have unique caches for each different amount that may be attempted on a noAmount invoice.

Reply:
Done https://github.com/GaloyMoney/galoy/pull/766/commits/fe518b1a8570a7a455fa1c1c71b436b85f3e2fc3